test/test_mkldnn.py

```diff
@@ -578,6 +578,28 @@ def test_transpose(self):
                     x.to_mkldnn().transpose(dim1, dim2).to_dense(),
                 )
 
+    def test_linear_non_contiguous_weight(self):
+        in_features = torch.randint(3, 10, (1,)).item()
+        out_features = torch.randint(3, 100, (1,)).item()
+        x = torch.randn(3, in_features, dtype=torch.float32) * 10
+        w = torch.randn(in_features, out_features, dtype=torch.float32)
+        # Only works for bias case by dispatch onednn linear to addmm.
+        for bias in [True]:
+            x1 = x.clone().requires_grad_()
+            x2 = x.clone().to_mkldnn().requires_grad_()
+            linear = torch.nn.Linear(in_features, out_features).float()
+            mkldnn_linear = copy.deepcopy(linear)
```

Comment:
We can first reinint the linear weight and then copy a module, such as:
```
linear = torch.nn.Linear(in_features, out_features).float()
linear.weight = torch.nn.Parameter(w.t())
mkldnn_linear = copy.deepcopy(linear)
```

Could you also enable **bias = Flase** test case and remove the comment in https://github.com/pytorch/pytorch/blob/4d703d040b3a08d2a9997bbdff1a58e8dcadedf6/test/test_mkldnn.py#L596

Reply:
Will do.
I didn't know deepcopy actually maintains the stride. I thought framework emits contiguous tensor whenever they create something fresh. :exploding_head: